anchore_engine/services/policy_engine/engine/vulns/mappers.py

```diff
@@ -16,38 +16,452 @@
 )
 
 
+class DistroMapper:
+    engine_distro = None
+    grype_os = None
+    grype_like_os = None
+
+    @staticmethod
+    def for_engine(distro: str):
+        return ENGINE_DISTRO_MAPPERS.get(distro)
+
+    @staticmethod
+    def for_grype(distro: str):
+        return GRYPE_DISTRO_MAPPERS.get(distro)
+
+    def to_grype_distro(self, version):
+        return {
+            "name": self.grype_os,
+            "version": version,
+            "idLike": self.grype_like_os,
+        }
+
+
+class RhelMapper(DistroMapper):
```

Comment:
These don't need to be classes if they don't have different logic. They should be the same type, just with parameters to the constructor. A child class expects to have different behavior. The state of an object is expected to differ per instance.

An alternative, and cleaner approach is something like:
mappers = {
"rhel": DistroMapper(engine_distro="rhel", grype_os="redhat", grype_like_os="fedora),
"debian": ...
}

Reply:
@zhill between rhel debian and apkg, mappers will have different  will have different logic for mapping source package to grype's artifact metadata